src/sentry/search/moresnuba/backend.py

```diff
@@ -0,0 +1,318 @@
+        if len({p.organization_id for p in projects}) != 1:
+            raise RuntimeError("Cross organization search not supported")
+
+        if paginator_options is None:
+            paginator_options = {}
+
+        group_queryset = Group.objects.filter(project__in=projects).exclude(
+            status__in=[
+                GroupStatus.PENDING_DELETION,
+                GroupStatus.DELETION_IN_PROGRESS,
+                GroupStatus.PENDING_MERGE,
+            ]
+        )
+
+        qs_builder_conditions = {
+            "bookmarked_by": QCallbackCondition(
+                lambda user: Q(bookmark_set__project__in=projects, bookmark_set__user=user)
+            ),
+            "assigned_to": QCallbackCondition(
+                functools.partial(assigned_to_filter, projects=projects)
+            ),
+            "unassigned": QCallbackCondition(
+                functools.partial(unassigned_filter, projects=projects)
+            ),
+            "subscribed_by": QCallbackCondition(
+                lambda user: Q(
+                    id__in=GroupSubscription.objects.filter(
+                        project__in=projects, user=user, is_active=True
+                    ).values_list("group")
+                )
+            ),
+        }
+
+        group_queryset = QuerySetBuilder(qs_builder_conditions).build(
+            group_queryset, search_filters
+        )
+        # filter out groups which are beyond the retention period
+        retention = quotas.get_event_retention(organization=projects[0].organization)
+        if retention:
+            retention_window_start = timezone.now() - timedelta(days=retention)
+        else:
+            retention_window_start = None
+        # TODO: This could be optimized when building querysets to identify
+        # criteria that are logically impossible (e.g. if the upper bound
+        # for last seen is before the retention window starts, no results
+        # exist.)
+        if retention_window_start:
+            group_queryset = group_queryset.filter(last_seen__gte=retention_window_start)
+
+        # This is a punt because the SnubaSearchBackend (a subclass) shares so much that it
+        # seemed better to handle all the shared initialization and then handoff to the
+        # actual backend.
+
+        results = self._query(
+            projects,
+            retention_window_start,
+            group_queryset,
+            environments,
+            sort_by,
+            limit,
+            cursor,
+            count_hits,
+            paginator_options,
+            search_filters,
+            date_from,
+            date_to,
+        )
+
+        end_time = time.time()
+        run_time = end_time - start_time
+        # print ("MoreSnuba Runtime: (seconds):", run_time)
+        # print ("MoreSnuba Results:", results.results)
+        logger.debug("SEN-1050/MoreSnuba Results: %r", results.results)
+        logger.debug("SEN-1050/MoreSnuba Runtime: (seconds): %r", run_time)
```

Comment:
It will be very hard to extract meaningful metrics from the logs. Why not:
1) bumping a datadog metric instead, which makes it easy to plot data
2) create an apm span so we can use our own apm to look into individual calls performance ?